### PR TITLE
feat: support certificate related options

### PR DESCRIPTION
**Pull request by kitsonk**
This PR enables support for TLS certificate related options for registry completions and caching remote dependencies which are being added to CLI.

Ref: denoland/deno#13467

### CODE REVIEW

**Review by dsherret — APPROVED** [review]
Almost LGTM. Just one comment.

File: package.json [hunk 1]
```diff
@@ -257,6 +266,21 @@
...
           }
         },
+        "deno.tlsCertificate": {
+          "type": "string",
+          "default": null,
+          "markdownDescription": "A path to a PEM certificate to use as the certificate authority when validating TLS certificates when fetching and caching remote resources. This is like using `--cert` on the Deno CLI and overrides the `DENO_CERT` environment variable if set.",
+          "scope": "window"
+        },
+        "deno.unsafelyIgnoreCertificateErrors": {
+          "type": "array",
```
[the remark this answers is not in the document]
Comment (kitsonk) [reply package.json:276]: No.

[See](https://github.com/denoland/deno/blob/main/cli/lsp/config.rs#L190-L193)

```rust
  /// An option, if set, will unsafely ignore certificate errors when fetching
  /// remote resources.
  #[serde(default)]
  pub unsafely_ignore_certificate_errors: Option<Vec<String>>,
```

I need to change the `.d.ts` files to `string[] | null`.

File: client/src/shared_types.d.ts [hunk 2]
```diff
@@ -43,9 +44,11 @@ export interface Settings {
       hosts: Record<string, boolean>;
     } | null;
   } | null;
+  tlsCertificate: string | null;
+  unsafelyIgnoreCertificateErrors: boolean;
   /** Determine if the extension should be type checking against the unstable
    * APIs. */
-  unstable: boolean;
+  unstable: string[] | null;
```
Comment (dsherret) [inline client/src/shared_types.d.ts:51]: Other one.

Comment (kitsonk) [reply client/src/shared_types.d.ts:51]: 🤦

**Pull request merged**